### PR TITLE
Archive all Notifications within the context of the page

### LINKED ISSUE #75: Archive all should be context specific
If I am viewing all issues from a specific repository, I want to clear them all in one click.
Archive all hits everything in all my notifications.

Is there currently a way to do what I want?

### PR DESCRIPTION
**Pull request by tarebyte**
Closes #75

This add the URL parameters to the Archive all post action so that we only archive in the context of what we are seeing.

I'm not super happy with this solution as I think we can DRY this up a bit, so I'd ❤️  feedback.

I also removed the Archive all button from the Archive page.

/cc @andrew

### CODE REVIEW

**Comment by andrew** [comment]
A good start to stop people accidentally archiving everything, feels like we can replace it with a "select all" checkbox and a single archive button the archives all the selected notifications

**Comment by tarebyte** [comment]
> A good start to stop people accidentally archiving everything, feels like we can replace it with a "select all" checkbox and a single archive button the archives all the selected notifications

Agreed, I don't think this is the end all solution. But in the mean time it will save users some confusion.

**Comment by andrew** [comment]
Merge away!

File: app/views/notifications/index.html.erb [hunk 1]
```diff
@@ -56,8 +56,20 @@
           <%= octicon 'sync', :height => 16 %>
         <% end %>
 
-        <% if @notifications.any? %>
-          <%= link_to 'Archive all', archive_notifications_path, class: 'btn btn-default', method: :post %>
+        <% if @notifications.any? && !params[:archive].present? %>
+          <%= link_to(
+                'Archive all',
```
Comment (briannelson) [inline app/views/notifications/index.html.erb]: @tarebyte change wording to be more clear?

"Archive this list" ?

**Comment by tarebyte** [comment]
@briannelson I've updated the view so that if are actually archiving all of your notifications it will say so. Otherwise it will say `Archive this list`. Thanks for the suggestion!

**Comment by andrew** [comment]
👍

**Pull request merged**